server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/globalstate/TransportSnapshotGlobalStateAction.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.admin.cluster.snapshots.globalstate;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.master.TransportMasterNodeAction;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.block.ClusterBlockException;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.snapshots.SnapshotsService;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+public class TransportSnapshotGlobalStateAction extends TransportMasterNodeAction<SnapshotGlobalStateRequest, SnapshotGlobalStateResponse> {
+    private final SnapshotsService snapshotsService;
+
+    @Inject
+    public TransportSnapshotGlobalStateAction(
+        TransportService transportService,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ActionFilters actionFilters,
+        IndexNameExpressionResolver indexNameExpressionResolver,
+        SnapshotsService snapshotsService
+    ) {
+        super(
+            SnapshotGlobalStateAction.NAME,
+            transportService,
+            clusterService,
+            threadPool,
+            actionFilters,
+            SnapshotGlobalStateRequest::new,
+            indexNameExpressionResolver,
+            SnapshotGlobalStateResponse::new,
+            ThreadPool.Names.SNAPSHOT_META
```

Comment:
No need to do the initial coordination work on `SNAPSHOT_META`, we just call `getRepositoryData` which forks again:

```suggestion
            ThreadPool.Names.SAME // immediately forks again to get the repository data
```

Reply:
Changed it to `SAME`.